### PR TITLE
 Added code of Coding Exercise-21

### PR DESCRIPTION
**Pull request by anushreeSonak**

### CODE REVIEW

File: First/dailyExercises/Permutations.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            for (int index = 0; index < suffix.length(); index++) {
+                String newPrefix = prefix + suffix.charAt(index);
+                String newSuffix = suffix.substring(0, index) + suffix.substring(index + 1);
+                permute(newPrefix, newSuffix, permutations);
+            }
+        }
+    }
+
+    public static void main(String[] args) {
+        String input = "anu";
```
Comment (waghmaresuyog) [inline First/dailyExercises/Permutations.java]: give meaningful name to variable name input

File: First/dailyExercises/Permutations.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package firstPackage.First.dailyExercises;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Permutations {
+
+    public static List<String> findPermutations(String input) {
+        List<String> permutations = new ArrayList<>();
```
Comment (waghmaresuyog) [inline First/dailyExercises/Permutations.java]: give different name from class

File: First/dailyExercises/Permutations.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        return permutations;
+    }
+
+    private static void permute(String prefix, String suffix, List<String> permutations) {
+        if (suffix.length() == 0) {
+            permutations.add(prefix);
+        } else {
+            for (int index = 0; index < suffix.length(); index++) {
+                String newPrefix = prefix + suffix.charAt(index);
+                String newSuffix = suffix.substring(0, index) + suffix.substring(index + 1);
```
Comment (waghmaresuyog) [inline First/dailyExercises/Permutations.java]: change the variable name

**Review by waghmaresuyog — CHANGES_REQUESTED** [review]

File: First/dailyExercises/Permutations.java (hunk 1 above)
Comment (anushreeSonak) [reply First/dailyExercises/Permutations.java]: Done !!

File: First/dailyExercises/Permutations.java (hunk 3 above)
Comment (anushreeSonak) [reply First/dailyExercises/Permutations.java]: done !

File: First/dailyExercises/Permutations.java (hunk 2 above)
Comment (anushreeSonak) [reply First/dailyExercises/Permutations.java]: done !

**Review by waghmaresuyog — APPROVED** [review]

**Review by prashantkawa — APPROVED** [review]
Approving PR

**Pull request merged**